kimchi/src/circuits/polynomials/foreign_mul/constraints.rs

```diff
@@ -0,0 +1,40 @@
+//!
+//! Fundamental constraints and related helpers
+//!
+
+use std::cell::RefCell;
+
+use ark_ff::{FftField, One, Zero};
+
+use crate::circuits::expr::{Cache, ConstantExpr, Expr, E};
+
+thread_local! {
+    static CACHE: std::cell::RefCell<Cache>  = RefCell::new(Cache::default());
+}
+
+pub(crate) fn _cache<F: FftField>(mut x: E<F>) -> E<F> {
+    CACHE.with(|cache| x = cache.borrow_mut().cache(x.clone()));
+    x
+}
```

Comment:
I guess the cache and this function are not used anywhere?

Reply:
maybe a good idea to delete it here? So I can review its actual usage :D this looks weird to me